packages/shared/src/components/profile/Readme.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {
+  FormEventHandler,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useState,
+} from 'react';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import request, { ClientError } from 'graphql-request';
+import { PublicProfile } from '../../lib/user';
+import AuthContext from '../../contexts/AuthContext';
+import { MyProfileEmptyScreen } from './MyProfileEmptyScreen';
+import Markdown from '../Markdown';
+import MarkdownInput from '../fields/MarkdownInput';
+import { UPDATE_README_MUTATION, USER_README_QUERY } from '../../graphql/users';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { graphqlUrl } from '../../lib/config';
+import { useToastNotification } from '../../hooks';
+import { formToJson } from '../../lib/form';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/ButtonV2';
+
+export interface ReadmeProps {
+  user: PublicProfile;
+}
+
+export function Readme({ user }: ReadmeProps): ReactElement {
+  const { displayToast } = useToastNotification();
+  const client = useQueryClient();
+  const { user: loggedUser } = useContext(AuthContext);
+  const isSameUser = loggedUser?.id === user.id;
+  const readme = user?.readmeHtml;
+
+  // Markdown is supported only in the client due to sanitization
+  const isClient = typeof window !== 'undefined';
+
+  const [editMode, setEditMode] = useState(false);
+
+  const queryKey = generateQueryKey(RequestKey.Readme, user);
+  const { data: remoteReadme, isLoading } = useQuery<{
```

Comment:
Fetches the readme as markdown only when needed. Lessons learned from the comments

Reply:
Optional could have had these in a useProfile hook or something, but not blocking the PR